tests/legacy/test_read_exceptions.py

```diff
@@ -149,14 +146,20 @@ def test_timeout_stream(self, generate_task, real_x_series_device):
         ):
             pytest.skip("Requires a plugin device.")
 
+        number_of_channels = 4
```

Comment:
not all `real_x_series` devices have 4 AO channels. Some only have 2, and a few have 0. If possible, I'd reduce this to 2 and skip if its 0. If that doesn't work, then skip if not 4 is fine, too.

Reply:
4 is a much better demonstration of how DAQmx packs the samples together.

How about using the `Device` object to query how many AO channels are supported? call `pytest.skip(reason)` if 0.

Also skip if `device.ai_simultaneous_sampling_supported` because those devices have a different way to program AO loopback (acquire from Dev1/ai0:3 with `ai_input_src="_ao0_vs_aognd"` etc.)